### DISCUSSION TITLE
[Options][File Types]: add warning about losing changes when "Apply" button hasn't been pressed

### DISCUSSION THREAD

**Discussion by PetbkA** [doublecmd/doublecmd#1078] [Suggestions]
### Description

Double Commander
Version: 1.0.10
Revision: 239
Commit: d00f8b7b7
Build date: 2023/01/21
Lazarus: 2.2.5.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 10 Home 22H2 x86_64

1. Changes in [_Options → File Types_](http://doublecmd.github.io/doc/en/configuration.html#ConfigColorFiles) is being saved if you press _Apply_ in _File Types_ frame. I.e. you can't add some category masks to some file type and save it by pressing _Apply_ or _OK_ from _Options_ menu - changes will be lost without warning;
2. Moreover, if you change some file type → Forget to press _Apply_ on _File Types_ frame and switch to other file type, you won't get any warning about losing your changes, although they will be lost (here's the [video](https://user-images.githubusercontent.com/22843816/216828922-95362a78-793f-423d-9353-16b3510e504b.mp4)).

### Proposed technical implementation details (optional)

I suggest to add warning about losing changes if _Apply_ in _File Types_ hasn't been pressed

**Comment by dandv** [comment]
Hi, I also just lost a bunch of file extensions I had typed, because I clicked "Add", thinking I would <kbd>Add</kbd> add that category:

![image](https://github.com/user-attachments/assets/112e7b2b-6b05-44cc-ab54-dde760fd56fa)

I think better UX can be done here. I'm not a UX expert, I can just tell what's not good UX :) but a good idea would be to steal this UX from another file manager.

**Comment by antodc** [comment]
I agree, and I already raised a similar point (see first question in #1955). Without luck, unfortunately.

**Comment by alexx2000** [comment]
I removed additional `Apply` button.
